pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -235,48 +232,59 @@ func MaintainMachineImages(shootLogger *logrus.Entry, shoot *gardencorev1beta1.S
 
 		shoot.Spec.Provider.Workers[i].Machine.Image = updatedMachineImage
 
-		message := fmt.Sprintf("image of worker-pool '%s' from '%s' version '%s' to version '%s'. Reason: %s", worker.Name, workerImage.Name, *workerImage.Version, *updatedMachineImage.Version, *reason)
+		message := fmt.Sprintf("image of worker-pool %q from %q version %q to version %q. Reason: %s", worker.Name, workerImage.Name, *workerImage.Version, *updatedMachineImage.Version, *reason)
 		reasonsForUpdate = append(reasonsForUpdate, message)
 	}
 
 	return reasonsForUpdate, nil
 }
 
-// MaintainKubernetesVersion determines if a shoots kubernetes version has to be maintained and in case returns the target version
-func MaintainKubernetesVersion(shoot *gardencorev1beta1.Shoot, profile *gardencorev1beta1.CloudProfile, logger *logrus.Entry) (updatedKubernetesVersion *string, messageKubernetesUpdate *string, error error) {
+// MaintainKubernetesVersion updates a Shoot's Kubernetes version if necessary and returns the reason why an update was done
+func MaintainKubernetesVersion(shoot *gardencorev1beta1.Shoot, profile *gardencorev1beta1.CloudProfile) (*string, error) {
 	shouldBeUpdated, reason, isExpired, err := shouldKubernetesVersionBeUpdated(shoot, profile)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
+	}
+	if !shouldBeUpdated {
+		return nil, nil
+	}
+
+	updatedKubernetesVersion, err := determineKubernetesVersion(shoot, profile, isExpired)
+	if err != nil {
+		return nil, err
+	}
+	if updatedKubernetesVersion == nil {
+		return nil, nil
+	}
+	reasonForKubernetesUpdate := fmt.Sprintf("Kubernetes version %q to version %q. Reason: %s", shoot.Spec.Kubernetes.Version, *updatedKubernetesVersion, *reason)
+	shoot.Spec.Kubernetes.Version = *updatedKubernetesVersion
+	return &reasonForKubernetesUpdate, err
```

Comment:
You could think about removing all of these string pointers. You could simply use the empty string instead of `nil` for distinguishing between there is a reason and there is no reason.

Reply:
done, thanks!